libs/langchain/langchain/llms/azureml_endpoint.py

```diff
@@ -233,6 +280,49 @@ class AzureMLOnlineEndpoint(LLM, BaseModel):
     model_kwargs: Optional[dict] = None
     """Keyword arguments to pass to the model."""
 
+    @root_validator()
+    @classmethod
+    def validate_endpoint_api_type(cls, values: Dict) -> AzureMLBaseEndpoint:
```

Comment:
I am wondering, is there any way that we should be checking that the url matches the class? As in a url with "v1/chat/completions" should only be used in the AzureMLChatOnlineEndpoint class and vice versa?

Reply:
@gdyre it's a good question. I was wondering the same thing. We can do that check for MaaS since URLs are different. We can't for MaaP. I don't see any point on using a chat API for a completions or vice versa. We can also warn the user instead of failing. Which one should we follow?